github-diff-file-toggle.user.js

```diff
@@ -25,29 +25,32 @@
 
 	const setToggleStyle = state => {
 		const mainToggle = $(".ghdt-toggle");
-		mainToggle.classList.toggle("ghdt-selected", state);
-		mainToggle.style = state
-			? "background-color: var(--color-btn-selected-bg);"
-			: "";
+		if (mainToggle) {
+			mainToggle.classList.toggle("ghdt-selected", state);
+			mainToggle.style = state ? "background-color: var(--color-btn-selected-bg);" : "";
+		}
 	};
 
-	const buildButton = () => {
+	const init = () => {
 		if (!$(".ghdt-toggle")) {
-			const button = make({
+			const toggleButton = make({
 				el: "button",
 				className: "btn btn-sm ghdt-toggle tooltipped tooltipped-s float-right",
 				text: "Toggle viewed",
 				attrs: {
 					"aria-label": "Toggle all viewed files"
 				}
 			});
-			on(button, "click", event => {
+			on(toggleButton, "click", event => {
 				toggle(document, !event.target.classList.contains("ghdt-selected"));
 			});
-			$("#files.diff-view")?.prepend(button);
+			const diffBarItem = make({
+				el: "div",
+				className: "diffbar-item js-batched-reviewed mr-3",
+			}, [toggleButton]);
+			$("diff-layout > div.pr-toolbar > div.diffbar > div.pr-review-tools")?.prepend(diffBarItem);
```

Comment:
This is what I did at first and then I changed it due to various reasons. Primarily, I didn't want to change the default layout since you know, accessibility, muscle memory and stuff! Next reason is the placement and structure of the primary HTML elements. I looked at the HTML source of the historical pages and found that inserting our element here unlikely to cause a break by github folks in the long run.

I know it looks good and I too prefer your way but remember this is not an official feature so tight integration will be difficult to maintain and will become an annoyance for users who depend on it.